Distraction Shield/background.js

```diff
@@ -0,0 +1,111 @@
+//Set that holds the urls to be intercepted
+var blockedSites = new BlockedSiteList();
+var interceptDateList = [];
+var localSettings = new UserSettings();
+
+/* --------------- ------ setter for local variables ------ ---------------*/
+
+setLocalSettings = function(newSettings) {
+    var oldState = localSettings.getState();
+    localSettings.copySettings(newSettings);
+    if (oldState != localSettings.getState()) {
+        replaceListener();
+    }
+};
+
+setLocalBlacklist = function(newList) {
+    blockedSites.setList(newList.getList());
+    replaceListener();
+};
+
+setLocalInterceptDateList = function(dateList) {
+    interceptDateList = dateList;
+};
+
+/* --------------- ------ Storage retrieval ------ ---------------*/
+// Methods here are only used upon initialization of the session.
+// Usage found in init.js
+
+retrieveSettings = function(callback, param) {
+    storage.getSettings(function(settingsObject) {
+        localSettings = settingsObject;
+        return callback(param);
+    });
+};
+
+retrieveBlockedSites = function(callback){
+    storage.getBlacklist(function(blacklist) {
+        blockedSites.setList(blacklist.getList());
+        return callback();
+    });
+};
+
+/* --------------- ------ Updating of variables ------ ---------------*/
+
+addUrlToBlockedSites = function(unformattedUrl, onSuccess) {
+    blockedSiteBuilder.createNewBlockedSite(unformattedUrl, function(newBS) {
+        if (blockedSites.addToList(newBS)) {
+            synchronizer.syncBlacklist(blockedSites);
+            onSuccess();
+        } 
+    });
+};
+
+/* --------------- ------ webRequest functions ------ ---------------*/
+
+replaceListener = function() {
+    removeWebRequestListener();
+    var urlList = blockedSites.getActiveUrls();
+    if (localSettings.getState() == "On" && urlList.length > 0) {
```

Comment:
Well, think about it this way: localSettings is an object of the UserSettings class. What you are doing is extracting the field status.state and performing a check on it. The idea behind using objects is to abstract from the data and how it is represented and encapsulate it in objects. The object is the manager of that data and when you use that object, you query it via functions. Best case is you should not care how the data is represented if a function can provide you with the appropriate needed response, thus, the boolean function in this case not only is a better way of coding (oop style) but it actually makes more sense, while having the string "On" or "Off" requires me to be aware of what the UserSettings class contains inside. If these things can be avoided, do avoid them, it improves clarity by a lot.